src/sentry/api/helpers/group_index.py

```diff
@@ -207,3 +225,99 @@ def validate(self, attrs):
             raise serializers.ValidationError(
                 'Other attributes cannot be updated when discarding')
         return attrs
+
+
+def handle_discard(request, group_list, projects, user, view_cls):
+    for project in projects:
+        if not features.has('projects:discard-groups', project, actor=user):
+            return Response({'detail': ['You do not have that feature enabled']}, status=400)
+
+    # grouped by project_id
+    groups_to_delete = defaultdict(list)
+
+    for group in group_list:
+        with transaction.atomic():
+            try:
+                tombstone = GroupTombstone.objects.create(
+                    previous_group_id=group.id,
+                    actor_id=user.id if user else None,
+                    **{name: getattr(group, name) for name in TOMBSTONE_FIELDS_FROM_GROUP}
+                )
+            except IntegrityError:
+                # in this case, a tombstone has already been created
+                # for a group, so no hash updates are necessary
+                pass
```

Comment:
This isn't a bug in your code, but I wonder if there's a potential bug here... What happens if we're running through this, we create a `GroupTombstone` and commit the transaction, but then something happens to the process, or an error occurs and we don't end up calling `delete_groups`? Will the groups just hang around forever?

Reply:
good point, but i'm not totally sure how to resolve given i don't think we're going to want to move away from doing async group deletions. do you have any suggestions?

Reply:
I don't think we need to tackle it in your refactor, just noticed it and thought it was worth mentioning. 

I had a couple of ideas on it, but none of them seem like they'd work, and I think fixing it properly might be a reasonable amount of effort. Maybe we should just document it to be fixed at some point?

Reply:
👍 APP-1018 if you want to add any of your ideas